gc.c

```diff
@@ -2659,6 +2661,19 @@ rb_gc_mark_roots(void *objspace, const char **categoryp)
 
 #define TYPED_DATA_REFS_OFFSET_LIST(d) (size_t *)(uintptr_t)RTYPEDDATA(d)->type->function.dmark
 
+void
+rb_gc_ractor_moved(VALUE dest, VALUE src)
+{
+    void *objspace = rb_gc_get_objspace();
+    if (UNLIKELY(FL_TEST_RAW(src, FL_SEEN_OBJ_ID))) {
+        rb_gc_impl_object_id_move(objspace, dest, src);
+    }
```

Comment:
I suppose the big argument for not transferring the `object_id` is that we'd save on locking the VM.

Reply:
I don't think ID should be inherited (moved object should be different object).

Reply:
I did it because to me it seemed that semantically, `move: true` means we're just changing which ractor owns the object, and not allocating a new object.

But if you think it shouldn't be preserved, it's fine by me, I don't think it's very importan.

Removed.